client/mysql.cc

```diff
@@ -1732,7 +1732,7 @@ static struct my_option my_long_options[] =
    GET_NO_ARG, NO_ARG, 0, 0, 0, 0, 0, 0},
   {"vertical", 'E', "Print the output of a query (rows) vertically.",
    &vertical, &vertical, 0, GET_BOOL, NO_ARG, 0, 0, 0, 0, 0, 0},
-  {"wait", 'w', "Wait and retry if connection is down.", 0, 0, 0, GET_NO_ARG,
+  {"wait", 'w', "Wait and retry if connection is down.", &wait_flag, 0, 0, GET_BOOL,
```

Comment:
that's not how it used to work. see commit c9d04284c8e (wait time was configurable, retrying forever)

and it's not how `-w` works in `mariadb-admin` (wait time is hard-coded, 5 seconds, number of retries is configurable)

For consistency, I suppose, it should do what `mariadb-admin` does